In [1]:
import pandas as pd
from scipy import stats
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# nilai matematika yang ikut bimbel
treatment = [85.2, 81.3, 75.3, 90, 71.2, 84.9, 95, 76, 70, 85.9]

# nilai matematika yang tidak ikut bimbel
control = [90.3, 71.1, 74.3, 78.9, 85.2, 81.1, 70, 67, 69, 50]


H0 = ikut bimbel atau tidak sama saja

H1 = ada perbedaan antara ikut bimbel dan tidak ikut bimbel

------
H0 : mu1 = mu2 atau mu1 - mu2 =0

H1 : mu1 - mu2 != 0 

In [3]:
mu1 = np.mean(treatment)
mu2 = np.mean(control)

d = np.abs(mu1 - mu2)
d

7.790000000000006

Semakin besar d maka semakin kecil p valuenya, artinya semakin besar selisih antara treatment dan control maka h0 semakin dapat ditolak

In [4]:
concat = np.concatenate([treatment,control])
concat

array([85.2, 81.3, 75.3, 90. , 71.2, 84.9, 95. , 76. , 70. , 85.9, 90.3,
       71.1, 74.3, 78.9, 85.2, 81.1, 70. , 67. , 69. , 50. ])

In [5]:
diff = []
p = 100
np.random.seed(1)

for i in range(p):
    np.random.shuffle(concat)
    a = concat[:10].mean()
    b = concat[10:].mean()
    diff.append(abs(a-b))

kriteria diff = 7.9

In [6]:
p_crit = np.where(diff >= d, 1, 0)
p_crit.sum()

5

In [7]:
p_value = p_crit.sum()/p
p_value

0.05

In [8]:
df = pd.read_csv("covid_19_indonesia_time_series_all.csv")
df

Date Location ISO Code          Location  New Cases  New Deaths  \
0       3/1/2020             ID-JK       DKI Jakarta          2           0   
1       3/2/2020             ID-JK       DKI Jakarta          2           0   
2       3/2/2020               IDN         Indonesia          2           0   
3       3/2/2020             ID-RI              Riau          1           0   
4       3/3/2020             ID-JK       DKI Jakarta          2           0   
...          ...               ...               ...        ...         ...   
21754  12/2/2021             ID-SA    Sulawesi Utara          2           0   
21755  12/2/2021             ID-SB    Sumatera Barat          4           0   
21756  12/2/2021             ID-SS  Sumatera Selatan          4           0   
21757  12/2/2021             ID-SU    Sumatera Utara          4           0   
21758  12/3/2021               IDN         Indonesia        245           8   

       New Recovered  New Active Cases  Total Cases  Total Deaths  \
0                  0                 2           39            20   
1                  0                 2           41            20   
2                  0                 2            2             0   
3                  0                 1            2             0   
4                  0                 2           43            20   
...              ...               ...          ...           ...   
21754              0                 2        34715          1056   
21755              0                 4        89849          2152   
21756              0                 4        59937          3071   
21757              1                 3       106045          2889   
21758            328               -91      4257243        143858   

       Total Recovered  ...  Latitude New Cases per Million  \
0                   41  ... -6.204699                  0.18   
1                   41  ... -6.204699                  0.18   
2                    0  ... -0.789275                  0.01   
3                    3  ...  0.511648                  0.16   
4                   41  ... -6.204699                  0.18   
...                ...  ...       ...                   ...   
21754            33566  ...  1.259638                  0.76   
21755            87605  ... -0.850253                  0.72   
21756            56830  ... -3.216212                  0.49   
21757           103061  ...  2.191894                  0.27   
21758          4105680  ... -0.789275                  0.92   

       Total Cases per Million New Deaths per Million  \
0                         3.60                   0.00   
1                         3.78                   0.00   
2                         0.01                   0.00   
3                         0.33                   0.00   
4                         3.96                   0.00   
...                        ...                    ...   
21754                 13140.24                   0.00   
21755                 16279.22                   0.00   
21756                  7293.78                   0.00   
21757                  7129.13                   0.00   
21758                 16053.83                   0.03   

      Total Deaths per Million Total Deaths per 100rb Case Fatality Rate  \
0                         1.84                   0.18             51.28%   
1                         1.84                   0.18             48.78%   
2                         0.00                   0.00              0.00%   
3                         0.00                   0.00              0.00%   
4                         1.84                   0.18             46.51%   
...                        ...                    ...                ...   
21754                   399.71                  39.97              3.04%   
21755                   389.91                  38.99              2.40%   
21756                   373.71                  37.37              5.12%   
21757                   194.22  

## Single sample hypothesis testing

In [18]:
# mencari new active cases pada bulan november tahun 2020 dan 2021
kasus_2020 = df[(pd.DatetimeIndex(df['Date']).month == 11) &
  (pd.DatetimeIndex(df['Date']).year == 2020)]

kasus_2021 = df[(pd.DatetimeIndex(df['Date']).month == 11) &
  (pd.DatetimeIndex(df['Date']).year == 2021)]

Mencari rata - rata active new case pada bulan november di tahun 2020 dan 2021

In [32]:
mean2020 = np.abs(kasus_2020['New Cases'].mean())
mean2021 = np.abs(kasus_2021['New Cases'].mean())
print(f'rata - rata kasus baru setiap harinya di tahun 2020 adalah: {mean2020}')
print(f'rata - rata kasus baru setiap harinya di tahun 2021 adalah: {mean2021}')

rata - rata kasus baru setiap harinya di tahun 2020 adalah: 246.20666666666668
rata - rata kasus baru setiap harinya di tahun 2021 adalah: 23.21944177093359


Apakah penurunan kasus dari 246 sampai 23 adalah penurunan yang signifikan ?

Hypothesis pada kasus ini adalah

**H0: μ >= 246**

**H1: μ < 23**

In [47]:
t_stat,p_val = stats.ttest_1samp(kasus_2021['New Cases'], mean2020)
print('P-value:',p_val/2) #The p-value divided by 2 since the output is two-sided p-value
print('t-statistics:',t_stat)

P-value: 0.0
t-statistics: -99.73592216161683


p valuenya 0.0 yang berarti kurang dari alpha, artinya h0 di reject yang berarti penurunannya statistically significant. t test yang dihasilkan negatif berarti setiap tahunnya kasus covid baru semakin menurun

### One Sample Two Tailed